packages/core-plugin-api/src/translation/Translation.tsx

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2023 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import React from 'react';
+import { TranslationRef } from './TranslationRef';
+import { useTranslationRef } from './useTranslationRef';
+import {
+  CollapsedMessages,
+  NestedMessageKeys,
+  PluralKeys,
+  TranslationFunctionOptions,
+} from '../apis/definitions/TranslationApi';
+
+/**
+ * @alpha
+ */
+export function Translation<
```

Comment:
As mentioned on Discord, I think the design used for this component in i18next is probably better, i.e. a child render callback. It lets you collect a couple more translations in one place, use your own components, etc. It should also making types a lot easier to manage. https://react.i18next.com/latest/translation-render-prop

Only change I think we need to make on top of that is that you pass the `translationRef` as a prop to the `Translation` component, like we already do here.